prime-router/src/main/kotlin/fhirengine/translation/hl7/HL7Truncator.kt

```diff
@@ -0,0 +1,122 @@
+package gov.cdc.prime.router.fhirengine.translation.hl7
+
+import ca.uhn.hl7v2.model.Type
+import ca.uhn.hl7v2.util.Terser
+import gov.cdc.prime.router.common.StringUtilities.trimAndTruncate
+import gov.cdc.prime.router.fhirengine.translation.hl7.utils.HL7Constants
+import gov.cdc.prime.router.fhirengine.translation.hl7.utils.HL7Constants.HD_FIELDS_LOCAL
+import gov.cdc.prime.router.fhirengine.translation.hl7.utils.HL7Constants.HD_TRUNCATION_LIMIT
+import gov.cdc.prime.router.fhirengine.translation.hl7.utils.HL7Utils
+import kotlin.math.min
+
+class HL7Truncator {
```

Comment:
Much of this class is brought over wholesale from HL7Serializer with a slight clean up.